_rules/invalid-form-field-value-36b590.md

```diff
@@ -0,0 +1,214 @@
+---
+id: 36b590
+name: Error message describes invalid form field value
+rule_type: atomic
+description: |
+  This rule checks that text descriptions are provided when the user completes a form field with invalid values or using an invalid format.
+accessibility_requirements:
+  wcag20:3.3.1: # Error Identification (A)
+    forConformance: true
+    failed: not satisfied
+    passed: further testing needed
+    inapplicable: further testing needed
+input_aspects:
+  - DOM Tree
+  - CSS Styling
+  - Language
+acknowledgements:
+  authors:
+    - Carlos Duarte
+    - João Vicente
+---
+
+## Applicability
+
+The rule applies to each HTML element with a [semantic role][] that inherits from the [abstract][] `input` or `select` roles.
+
+## Expectation 1
+
+For each test target where the user input caused the appearance of an [error message][], the error message is [visible][] and [included in the accessibility tree][].
+
+## Expectation 2
+
+Each [error message][] from Expectation 1 includes the identification the test target.
+
+## Expectation 3
+
+Each [error message][] from Expectation 1 describes the error.
+
+## Assumptions
+
+_There are currently no assumptions_
+
+## Accessibility Support
+
+_There are no major accessibility support issues known for this rule._
+
+## Background
+
+- [Understanding Success Criterion 3.3.1: Error Identification](https://www.w3.org/WAI/WCAG21/Understanding/error-identification)
+- [G84: Providing a text description when the user provides information that is not in the list of allowed values](https://www.w3.org/WAI/WCAG21/Techniques/general/G84)
+- [G85: Providing a text description when user input falls outside the required format or values](https://www.w3.org/WAI/WCAG21/Techniques/general/G85)
+
+## Test Cases
+
+### Passed
+
+#### Passed Example 1
+
+The error message provided in the vicinity of the `input` element identifies the input field (via its label) and describes the error.
+
+```html
+<script>
+	function processForm() {
+		document.getElementById('error').innerText = ''
+		var age = document.getElementById('age').value
+		console.log(isNaN(age))
+		if (!age || isNaN(age)) {
+			document.getElementById('error').innerText = 'Age must be a number'
+		} else if (age < 1) {
+			document.getElementById('error').innerText = 'Age must be at least 1'
+		}
+	}
+</script>
+
+<form>
+	<label for="age">Age (years)</label>
+	<input type="number" id="age" />
+	<span id="error"></span><br />
+	<input type="button" value="Submit" onclick="processForm()" />
+</form>
+```
+
+### Failed
+
+#### Failed Example 1
+
+No error message is provided.
+
+```html
+<form>
+	<label for="age">Age (years)</label>
+	<input type="number" id="age" />
+	<input type="button" value="Submit" />
+</form>
+```
+
+#### Failed Example 2
+
+The error message does not identify the input field.
```

Comment:
Not quite convinced the fact that the error here doesn't explicitly name the input in question is a failure. The proximity to the field that is in error should be sufficient here. Otherwise, we'd have to hard-fail any error message that doesn't explicitly identify the input field?

Perhaps a better example would be a form with 2 or more inputs, and a generic "You have errors, go and fix them" error message at the start of the form?

Reply:
Indeed. I'll updated the example following your suggestion.